lib/services/swagger-api-service.js

```diff
@@ -356,6 +355,8 @@ function swaggerFactory(
                 .then(function (activeTask) {
                     return activeTask.taskId;
                 });
+        } else {
```

Comment:
Just a minor suggestion, but I think this would be a little easier to read if written like:
```
return Promise.try(function() {
    if (nodeId) {
        return taskProtocol.activeTaskExists(nodeId)
        .then(function (activeTask) {
             return activeTask.taskId;
        });
});
```